packages/engine/Source/DataSources/parseGeoJson.js

```diff
@@ -0,0 +1,496 @@
+  if (!defined(geometryHandler)) {
+    throw new RuntimeError(`Unknown geometry type: ${geometryType}`);
+  }
+  return geometryHandler(feature, feature.geometry, crsFunction, options);
+}
+
+function processFeatureCollection(
+  featureCollection,
+  notUsed,
+  crsFunction,
+  options,
+) {
+  const features = featureCollection.features;
+  return features.map((feature) =>
+    processFeature(feature, undefined, crsFunction, options),
+  );
+}
+
+// function load(that, geoJson, options, sourceUri, clear) {
+//   let name;
+//   if (defined(sourceUri)) {
+//     name = getFilenameFromUri(sourceUri);
+//   }
+
+//   if (defined(name) && that._name !== name) {
+//     that._name = name;
+//     that._changed.raiseEvent(that);
+//   }
+
+//   const typeHandler = geoJsonObjectTypes[geoJson.type];
+//   if (!defined(typeHandler)) {
+//     throw new RuntimeError(`Unsupported GeoJSON object type: ${geoJson.type}`);
+//   }
+
+//   //Check for a Coordinate Reference System.
+//   const crs = geoJson.crs;
+//   let crsFunction = crs !== null ? defaultCrsFunction : null;
+
+//   if (defined(crs)) {
+//     if (!defined(crs.properties)) {
+//       throw new RuntimeError("crs.properties is undefined.");
+//     }
+
+//     const properties = crs.properties;
+//     if (crs.type === "name") {
+//       crsFunction = crsNames[properties.name];
+//       if (!defined(crsFunction)) {
+//         throw new RuntimeError(`Unknown crs name: ${properties.name}`);
+//       }
+//     } else if (crs.type === "link") {
+//       let handler = crsLinkHrefs[properties.href];
+//       if (!defined(handler)) {
+//         handler = crsLinkTypes[properties.type];
+//       }
+
+//       if (!defined(handler)) {
+//         throw new RuntimeError(
+//           `Unable to resolve crs link: ${JSON.stringify(properties)}`,
+//         );
+//       }
+
+//       crsFunction = handler(properties);
+//     } else if (crs.type === "EPSG") {
+//       crsFunction = crsNames[`EPSG:${properties.code}`];
+//       if (!defined(crsFunction)) {
+//         throw new RuntimeError(`Unknown crs EPSG code: ${properties.code}`);
+//       }
+//     } else {
+//       throw new RuntimeError(`Unknown crs type: ${crs.type}`);
+//     }
+//   }
+
+//   return Promise.resolve(crsFunction).then(function (crsFunction) {
+//     if (clear) {
+//       that._entityCollection.removeAll();
+//     }
+
+//     // null is a valid value for the crs, but means the entire load process becomes a no-op
+//     // because we can't assume anything about the coordinates.
+//     if (crsFunction !== null) {
+//       typeHandler(that, geoJson, geoJson, crsFunction, options);
+//     }
+
+//     return Promise.all(that._promises).then(function () {
+//       that._promises.length = 0;
+//       DataSource.setLoading(that, false);
+//       return that;
+//     });
+//   });
+// }
+
+/**
+ *
+ * @param {GeoJson} geoJson
+ * @param {ParseOptions} [options]
+ * @returns {Entity | Entity[] | undefined}
```

Comment:
Do you think it would be possible to map this to a list of `Feature` objects? And that, in turn, those could be mapped to Entities in another function to be used `GeoJsonDataSource`? 

That way, the future implementation could map the returned features to primitives.